sale_product_pack/tests/test_sale_product_pack.py

```diff
@@ -98,7 +98,7 @@ def test_create_components_price_order_line(self):
         self.assertAlmostEqual(line.price_subtotal, 27.68)
         self.assertEqual(
             (self.sale_order.order_line - line).mapped("price_subtotal"),
-            [1755.0, 22.5, 885.0],
+            [1579.5, 20.25, 796.5],
```

Comment:
Why this change?

Reply:
Before this PR the pricelist in the sale orders didn’t affect the pack components, which IMO is a bug. In the test, the pricelist “discount_pricelist” applies a 10% discount on all the products so we should modify the expected the components price. 
For example one of the components is "[FURN_9001] Flipover".

1. The product list price is: 1950. 
2. With the pricelist discount (10%): 1950 * 0.9 = 1755
3. With the PACK discount (10%): 1755 * 0.9 = 1579.50

![Screenshot](https://github.com/OCA/product-pack/assets/58082586/15e30c4a-385e-48b2-b5c8-6b01a7b20d3a)